commit 4de344be943145dcd7bfc1c0bb87398bc35f9403
Author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date:   Tue Nov 2 19:20:34 2021 +0800

    onscreen/native: Warn instead of crashing if flipping an inactive CRTC
    
    This can happen after hotplugging during heavy animation.

diff --git a/src/backends/native/meta-onscreen-native.c b/src/backends/native/meta-onscreen-native.c
index 031b3c9a4..b39308e68 100644
--- a/src/backends/native/meta-onscreen-native.c
+++ b/src/backends/native/meta-onscreen-native.c
@@ -402,7 +402,7 @@ meta_onscreen_native_flip_crtc (CoglOnscreen                *onscreen,
   kms = meta_kms_device_get_kms (kms_device);
   kms_update = meta_kms_ensure_pending_update (kms, kms_device);
 
-  g_assert (meta_gpu_kms_is_crtc_active (gpu_kms, crtc));
+  g_return_if_fail (meta_gpu_kms_is_crtc_active (gpu_kms, crtc));
 
   renderer_gpu_data = meta_renderer_native_get_gpu_data (renderer_native,
                                                          render_gpu);
